tests/HttpExtractorTest.php

```diff
@@ -41,21 +41,25 @@ public function testExtractSavesResponseToFile(): void
         $this->assertSame('http://example.com/result.txt', (string)$request->getUri());
     }
 
-    public function testExtractThrowsExceptionFor404Response(): void
-    {
-        $resource = new Uri('http://example.com/result.txt');
-        $content = 'File contents';
-        $mockedResponse = new Response(404, [], $content);
+    /**
+     * @dataProvider provideUrlsAndExceptions
+     */
+    public function testHttpExceptions(
+        Response $mockedResponse,
+        string $exceptionClass,
+        string $exceptionMessagePart
+    ): void {
         $client = $this->getMockedGuzzle([$mockedResponse]);
         $extractor = new HttpExtractor($client);
-        $destination = tempnam(sys_get_temp_dir(), 'http_extractor');
+        $temp = new Temp(__CLASS__);
```

Comment:
Prefix vůbec není nutnej, ten snad slouží jen pro lepší orientaci ve filesystemu, když se něco debuguje.

Reply:
Nebo když hrozej kolize při `uniqid()`.

Reply:
Sám ho ale koukám používám, takže mě zmátlo jen to použití `__CLASS__` :-)